core/src/main/java/io/grpc/internal/DelayedClientTransport.java

```diff
@@ -250,6 +302,37 @@ int getPendingStreamsCount() {
     }
   }
 
+  /**
+   * True return value indicates that the delayed transport is in a back-off interval (in
+   * TRANSIENT_FAILURE), that all fail fast streams (including pending as well as new ones) should
+   * fail immediately, and that non-fail fast streams can be created as {@link PendingStream} and
+   * should keep pending during this back-off period.
+   */
+  boolean isInBackoffPeriod() {
+    synchronized (lock) {
+      return inBackoffPeriod;
+    }
+  }
+
+  /**
+   * Is only called at the beginning of {@link TransportSet#scheduleBackoff}.
+   */
+  void startBackoff() {
```

Comment:
I'd really like this to be combined with `clearFailFastPendingStreams()` to avoid misusages. Maybe just name it `startBackoff(Status)` and it sets `inBackoffPeriod = true` and clears any fail fast streams.